modules/kubernetes-addons/velero/locals.tf

```diff
@@ -0,0 +1,56 @@
+locals {
+  name                 = "velero"
+  service_account_name = local.name
+  s3bucketprefix = "eks-tf-velero-backup"
+  s3bucketname = aws_s3_bucket.s3.id
+
+  default_helm_config = {
+    name        = local.name
+    chart       = local.name
+    repository  = "https://vmware-tanzu.github.io/helm-charts/"
+    version     = "2.23.6"
+    namespace   = local.name
+    description = "Velero AddOn Helm Chart"
+    values      = local.default_helm_values
+  }
+
+  default_helm_values = [templatefile("${path.module}/values.yaml", {
+    velero-sa-name = local.service_account_name
+    s3-bucket-name = local.s3bucketname
+  })]
+
+  helm_config = merge(
+    local.default_helm_config,
+    var.helm_config
+  )
+
+  # Set serviceAccount.create to False explicity
+  # even if its set to true in customer provided values.yaml
+  set_values = [
+    {
+      name  = "serviceAccount.name"
+      value = local.service_account_name
+    }
+  ]
```

Comment:
Looking at https://github.com/vmware-tanzu/helm-charts/blob/6c365d5540f57e943b0dd50faf59a4664e2babbd/charts/velero/values.yaml#L299 it should be `serviceAccount.server.name".

`set_values` also needs `serviceAccount.server.create = false` to override the default since we are creating serviceAccount here.